common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -992,6 +1233,45 @@ AppShufflePartitionInfo getPartitionInfo() {
     }
   }
 
+  /**
+   * Simply encodes an application attempt ID.
+   */
+  public static class AppAttemptId {
```

Comment:
For the various json beans we have `AppAttemptId`, `AppAttemptShuffleMergeId`, `AppPathsInfo` - can we make sure the `equals`, `toString` and `hashCode` follow similar patterns ?

For example, `AppAttemptId`/`AppPathsInfo` is allowing for subclasses to be passed in, while `AppAttemptShuffleMergeId` check if the arg is specifically of the same class.

Also, in `equals`, go from cheapest check to more expensive ones progressively.

Reply:
Removed all the equals and hashcode for these Json beans classes. Is it required?

Reply:
Turns out that we have keep the equals, as in UT it needs to check whether two objects are the same.

Reply:
Yes, `equals` would be required.
I was suggesting to make sure they are all following similar pattern, and go from cheaper to more expensive checks.
Also, to have `hashCode` if we are overriding `equals`

Reply:
Added hashcode in the latest commit, and also updated the equals to start from cheaper one to more expensive ones.